### PR TITLE
[helper] Fix build_image --pull and refactor

### PR DESCRIPTION
**Pull request by jonathanmetzman**
Fix behavior of build_image --pull (Fixes https://github.com/google/oss-fuzz/issues/5640)

Also refactor helper.py:
1. Change behavior of functions so that most return True on success and False on failure.
Only `main` will return 1 on failure and 0 on success now. Previous behavior was very error prone.
2. Rename `_get_output_dir` to `_get_out_dir`.
3. Make function docstrings use descriptive tense.
4. Make helper.py print help when no argument is specified.

### CODE REVIEW

File: infra/helper.py [hunk 1]
```diff
@@ -80,34 +84,42 @@ def main():  # pylint: disable=too-many-branches,too-many-return-statements
...
+    result = reproduce(args)
+  elif args.command == 'shell':
+    result = shell(args)
+  elif args.command == 'pull_images':
+    result = pull_images()
+  else:
+    # Print help string if no arguments provided.
+    parser.print_help()
+    result = False
+  return bool_to_retcode(result)
```
Comment (inferno-chromium) [inline infra/helper.py:110]: can also do int(result)

**Review by inferno-chromium — APPROVED** [review]
Thanks a lot!

File: infra/helper.py (hunk 1 above)
Comment (jonathanmetzman) [reply infra/helper.py:110]: Not really.
`int(False)` is `0`.

File: infra/helper.py [hunk 2]
```diff
@@ -955,13 +962,13 @@ def generate(args):
     file_handle.write(templates.BUILD_TEMPLATE % template_args)
 
   os.chmod(build_sh_path, 0o755)
-  return 0
+  return False
```
[the remark this answers is not in the document]
Comment (jonathanmetzman) [reply infra/helper.py]: Yep. Good catch.

File: infra/helper.py [hunk 3]
```diff
@@ -80,34 +84,42 @@ def main():  # pylint: disable=too-many-branches,too-many-return-statements
...
+    result = pull_images()
+  else:
+    # Print help string if no arguments provided.
+    parser.print_help()
+    result = False
+  return bool_to_retcode(result)
+
+
+def bool_to_retcode(boolean):
+  """Returns True if |boolean| is 0, the standard return value for a successful
```
Comment (oliverchang) [inline infra/helper.py:114]: comment seems like it's the wrong way around.

File: infra/helper.py [hunk 4]
```diff
@@ -80,34 +84,42 @@ def main():  # pylint: disable=too-many-branches,too-many-return-statements
...
+  process execution. Returns False otherwise."""
+  return 0 if boolean is True else 1
```
Comment (oliverchang) [inline infra/helper.py:116]: do we actually need the "is True" ?

File: infra/build_specified_commit.py [hunk 5]
```diff
@@ -231,7 +231,7 @@ def cleanup():
                                        env_to_add=None,
                                        source_path=host_src_path,
                                        mount_location='/src')
-    if result == 0 or i == num_retry:
+    if result or i == num_retry:
```
Comment (oliverchang) [inline infra/build_specified_commit.py:234]: I think you need to fix line 288 as well. Please double check other usages that need to be changed as well.

**Review by oliverchang — APPROVED** [review]
needs some minor fixes but overall thanks for doing this!

File: infra/build_specified_commit.py (hunk 5 above)
Comment (jonathanmetzman) [reply infra/build_specified_commit.py:234]: Done.

**Pull request merged**